### DISCUSSION TITLE
Extend "Marketplace tutorial" in the way that each vendor can have multi stores and payment related to stores

### DISCUSSION THREAD

**Discussion by foxpcteam** [adrien2p/medusa-extender#78] [Q&A] [answered]
Hi, 
Thanks for your great work on medusa-extender.

It will be great to see some example to extend "Marketplace tutorial" in the way that each vendor can have multi stores as in the real world.

* by the way, multi stores can also work as "Branches"

As for now, "Marketplace tutorial" example will allow to each(user) have only one shop. In this way, vendor have to open new user for every shop and this bad user experience.

Instant vendor(user) can create as many shops as he needs from one user and just change the shop from admin.

**Comment by adrien2p** [comment, marked as answer]
Hi there, if you want to support multi store for each user you just need to create a many to many relationship between user and store and then you can update anything related in order to load the current store for the logged in user ☺️

**Reply by foxpcteam** [reply]
Hi @adrien2p, thanks for your answer. Yes, this is what i have done.
Now another important question.
How to Handel multi payment configuration? Some vendors will ask to use their own stripe/PayPal or any other payment account to charge the client.
And then, we as the website owners can charge the vendor based on his "package".
Now as one owner site, we save this parameters inside medusa-config.js, but what will be the right way to make this work with multi vendors.

**Reply by adrien2p** [reply]
One way I think of now, which does not means the best approach since it needs some thinking, would be that when a payment session is created on your backend you add some metadata or a new fields that hold a segregation value such as the store from where the session comes from. Once that is done, you can create a new payment provider on your project, see the medusa doc, (let says that it extends the stripe provider) that you can use instead of the original stripe plugin, then in that extended provider you can manage the payment based on the data you have stored on the payment session and eventually the configuration you have asked the owner to setup on his profile. Hope that I am clear
